### ISSUE TITLE
can't get webhook-templates.xml to work

### ISSUE THREAD

**Issue by aaronjensen** [tcplugins/tcWebHooks#45]
I put this in my `config/webhook-templates.xml` and I don't see any new templates in my template selector. I'm on tcWebHooks version: 1.1.109.134

```xml
<?xml version="1.0" encoding="UTF-8"?>
<webhook-templates>
  <webhook-template enabled="true" name="slack" rank="30">
    <template-description>Slack Integration</template-description>
    <template-tool-tip value="Slack"/>

    <formats>
      <format name="jsontemplate" enabled="true" />
    </formats>

    <templates>
      <template>
        <template-text><![CDATA[
          {
            "attachments": [
              {
                "fallback": "${buildName} <${buildStatusUrl}|build #${buildNumber}> was successful",
                "text": "${buildName} <${buildStatusUrl}|build #${buildNumber}> was successful.",
                "color": "good"
              }
            ]
          }
        ]]></template-text>

        <branch-template-text><![CDATA[
          {
            "attachments": [
              {
                "fallback": "${buildName} <${buildStatusUrl}|build #${buildNumber}> branch: ${branchName} was successful",
                "text": "${buildName} <${buildStatusUrl}|build #${buildNumber}> branch: ${branchName} was successful.",
                "color": "good"
              }
            ]
          }
        ]]></branch-template-text>

        <states>
          <!-- <state type="buildBroken" enabled="true" /> -->
          <!-- <state type="changesLoaded" enabled="true" /> -->
          <!-- <state type="buildInterrupted" enabled="true" /> -->
          <!-- <state type="buildFixed" enabled="true" /> -->
          <!-- <state type="buildFinished" enabled="true" /> -->
          <!-- <state type="buildStarted" enabled="true" /> -->
          <!-- <state type="responsibilityChanged" enabled="true" /> -->
          <state type="buildSuccessful" enabled="true" />
          <!-- <state type="buildFailed" enabled="true" /> -->
          <!-- <state type="beforeBuildFinish" enabled="true" /> -->
        </states>
      </template
      <template>
        <template-text><![CDATA[
          {
            "attachments": [
              {
                "fallback": "${buildName} <${buildStatusUrl}|build #${buildNumber}> failed",
                "text": "${buildName} <${buildStatusUrl}|build #${buildNumber}> failed",
                "color": "danger"
              }
            ]
          }
        ]]></template-text>

        <branch-template-text><![CDATA[
          {
            "attachments": [
              {
                "fallback": "${buildName} <${buildStatusUrl}|build #${buildNumber}> branch: ${branchName} failed",
                "text": "${buildName} <${buildStatusUrl}|build #${buildNumber}> branch: ${branchName} failed",
                "color": "danger"
              }
            ]
          }
        ]]></branch-template-text>

        <states>
          <!-- <state type="buildBroken" enabled="true" /> -->
          <!-- <state type="changesLoaded" enabled="true" /> -->
          <!-- <state type="buildInterrupted" enabled="true" /> -->
          <!-- <state type="buildFixed" enabled="true" /> -->
          <!-- <state type="buildFinished" enabled="true" /> -->
          <!-- <state type="buildStarted" enabled="true" /> -->
          <!-- <state type="responsibilityChanged" enabled="true" /> -->
          <!-- <state type="buildSuccessful" enabled="true" /> -->
          <state type="buildFailed" enabled="true" />
          <!-- <state type="beforeBuildFinish" enabled="true" /> -->
        </states>
      </template
    </templates>
  </webhook-template>
</webhook-templates>
```

**Comment by netwolfuk** [comment]
The file name needs to be `webhook-templates. xml` (not pack).

**Comment by aaronjensen** [comment]
@netwolfuk sorry, I had webpack on the brain. It is actually named `webhook-templates.xml`. I can't reopen since you closed it.

**Comment by netwolfuk** [comment]
Is TeamCity running as root??!! Probably won't be an issue. ;-) just my ocd kicking in.

Is there any logging when you edit the file? Is it in .BuildServer/config ? What version of the plugin are you running?

**Comment by aaronjensen** [comment]
@netwolfuk hah, it is, but it's in docker.

Version is: `1.1.109.134`

[2016-08-18 20:46:24,739]   INFO -   jetbrains.buildServer.SERVER - WebHookTemplateFileChangeHandler :: Handling change to file: /data/teamcity_server/datadir/config/webhook-templates.xml requested by FileWatcher
[2016-08-18 20:46:24,758]   WARN -   jetbrains.buildServer.SERVER - WebHookTemplateFileChangeHandler :: Exception occurred attempting to reload WebHookTemplates. Could not parse: /data/teamcity_server/datadir/config/webhook-templates.xml

ugh, I had a missing `>`. My bad 😢

**Comment by netwolfuk** [comment]
That's a surprisingly useful log message. :-) yey me.

Yeah. I really need to get that ui going. 

I set myself the goal of getting a REST api working and then building the template editing ui to use that. That was 18 months ago :-(

I've made good progress this week getting spring 4.x and jersey 1.x to play nice inside junit so now i can crack on with building out the REST api. 

And then a angular app to edit the templates. Unit tested JavaScript FTW. (I can dream).
